src/core.js

```diff
@@ -338,8 +338,34 @@ export default function Core(rootElement, userSettings, rootInstanceSymbol = fal
      * @param {boolean} [keepEmptyRows] Optional. Flag for preventing deletion of empty rows.
      */
     alter(action, index, amount = 1, source, keepEmptyRows) {
-      let delta;
+      const trimSelectionToTableSize = () => {
```

Comment:
Can I ask you to move the code to the `adjustRowsAndCols` (https://github.com/handsontable/handsontable/blob/feature/issue-7337-simplified/src/core.js#L707) method? There is already an implemented mechanism for cell selection. I think that there is the best place for extending that functionality.